src/components/Navigation.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect, useState } from 'react';
+import { Header, HeaderItem } from '@hex-labs/core';
+import axios from 'axios';
+import { signOut, getAuth } from 'firebase/auth';
+import { Link, useLocation, useNavigate } from 'react-router-dom';
+
+import { useAuth } from '../contexts/AuthContext';
+import { app } from "../util/firebase";
+
+const auth = getAuth(app);
+
+const Navigation: React.FC = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const { user } = useAuth();
+  const hexathonId = window.sessionStorage.getItem("hexathonId");
+  const [role, setRoles] = useState<any>({
+    member: false,
+    exec: false,
+    admin: false,
+  });
+
+  const logOut = async () => {
+    signOut(auth);
+    await axios.post("https://auth.api.hexlabs.org/auth/logout");
+    window.location.href = `https://login.hexlabs.org`;
+  };
+
+  useEffect(() => {
+    const getRoles = async () => {
+      const res = await axios.get(
+        `https://users.api.hexlabs.org/users/${user?.uid}`
+      );
+      setRoles({
+        member: res.data.member,
+        exec: res.data.exec,
+        admin: res.data.admin,
+      });
+    };
```

Comment:
The AuthContext should already provide the user roles, so you shouldn't need to make another api request for this.